localstack/services/logs/provider.py

```diff
@@ -341,20 +348,30 @@ def moto_put_log_events(self: "MotoLogStream", log_events):
             payload_gz_encoded = output.getvalue()
             event = {"awslogs": {"data": base64.b64encode(output.getvalue()).decode("utf-8")}}
 
+            arn_data = parse_arn(destination_arn)
+
             if ":lambda:" in destination_arn:
-                client = connect_to(region_name=extract_region_from_arn(destination_arn)).lambda_
+                client = connect_to(
+                    aws_access_key_id=arn_data["account"], region_name=arn_data["region"]
+                ).lambda_
                 lambda_name = arns.lambda_function_name(destination_arn)
                 client.invoke(FunctionName=lambda_name, Payload=json.dumps(event))
```

Comment:
Same as commented above, I believe we shouldn't extract the lambda function name but use the ARN to invoke the function:
```python
client = connect_to(region_name=extract_region_from_arn(destination_arn)).lambda_
client.invoke(FunctionName=destination_arn, Payload=json.dumps(event))
```